doc/user_guide/io.rst

```diff
@@ -418,6 +420,64 @@ Extra saving arguments
     saving a file, be aware that it may not be possible to load it in some platforms.
 
 
+.. _zspy-format:
+
+ZSpy - HyperSpy's Zarr Specification
+------------------------------------
+
+This is the an additional format which guarantees that no
+information will be lost in the writing process and that supports saving data
+of arbitrary dimensions. It is based on the `Zarr project <https://zarr.readthedocs.io/en/stable/index.html>`_. Which exists as a drop in
+replacement for hdf5 with the intention to fix some of the speed and scaling
+issues with the hdf5 format. **If you are working with very large datasets lazily
+we recommend using .zspy for saving and loading your data**
+
+.. code-block:: python
+
+    >>> s = hs.signals.BaseSignal([0])
+    >>> s.save('test.zspy') # will save in nested directory
+    >>> hs.load('test.zspy') # loads the directory
+
+
+When saving to ``zspy``, all supported objects in the signal's
+:py:attr:`~.signal.BaseSignal.metadata` is stored. This includes lists, tuples and signals.
+Please note that in order to increase saving efficiency and speed, if possible,
+the inner-most structures are converted to numpy arrays when saved. This
+procedure homogenizes any types of the objects inside, most notably casting
+numbers as strings if any other strings are present:
+
+Extra saving arguments
+^^^^^^^^^^^^^^^^^^^^^^
+
+- ``compressor``: A `Numcodecs Codec <https://numcodecs.readthedocs.io/en/stable/index.html?>`_.
+   A compresssor can be passed to the save function to compress the data efficiently. The defualt
+   is to call a Blosc Compressor object.
+
+.. code-block:: python
+
+    >>> from numcodecs import Blosc
+    >>> compressor=Blosc(cname='zstd', clevel=1, shuffle=Blosc.SHUFFLE)
+    >>> s.save('test.zspy', compressor = compressor) # will save with Blosc compression
+
+.. note::
+
+    Compression can significantly increase the saving speed. If file size is not
+    an issue, it can be disabled by setting ``compressor=None``. In general we recommend
+    compressing your datasets as it can greatly reduce i-o overhead
+
+- ``write_to_storage``: The write to storage option allows you to pass the path to a directory (or database)
```

Comment:
Yea I considered both of those possibilities but decided against them for kind of the reason you described above. 

For the passing of a store argument  your filename doesn't really mean anything as only the store argument will be passed. If I remember correctly this complicates things a little bit. 

The second one is pretty do-able and I think that I got it working at one point. I could be convinced to get that working again.  The question is if that really makes sense.

Reply:
Going for option 2 should be good because this is a pattern used by the [`zarr.group`](https://zarr.readthedocs.io/en/stable/api/hierarchy.html#zarr.hierarchy.group), as it can take a path or a zarr store as first argument.